pkg/asset/machines/worker.go

```diff
@@ -182,11 +182,22 @@ func defaultNutanixMachinePoolPlatform() nutanixtypes.MachinePool {
 	}
 }
 
-func awsDefaultMachineTypes(region string, arch types.Architecture) []string {
+func awsDefaultMachineTypes(region string, arch types.Architecture, singleNodeControlPlane bool) []string {
```

Comment:
Hmm. I wonder if it doesn't make more sense to do:
```suggestion
func awsDefaultMachineTypes(region string, arch types.Architecture, instanceSize string) []string {
```
and then pass different sizes for controlPlane single node, controlPlane multi node, and compute. Practically it makes no difference, but I find it more clear semantically.
That way you could define in `pkg/asset/machines/master.go`:
```
const (
    defaultInstanceSize string = "xlarge"
    defaultSingleNodeInstanceSize string = "2xlarge"
)

[...]

instanceSize := defaultInstanceSize
if pool.Replicas != nil && *pool.Replicas == 1 {
    instanceSize = defaultSingleNodeInstanceSize
}
[...]
awsDefaultMachineType(..., instanceSize)
```

and for the worker in `pkg/asset/machines/worker.go`:
```
const defaultInstanceSize string = "xlarge"

[...]

awsDefaultMachineTypes(..., defaultInstanceSize)
```

Reply:
I had to make some adjustments to your suggestion for two reasons:

* worker.go and master.go share the same package so they would have to have a different prefix given to them
* worker.go and master.go apply to all clouds, while these constants are AWS specific